pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -68,6 +69,17 @@ func (ct *connTrackSystem) DumpFlows(zoneFilter uint16) ([]*flowexporter.Connect
 		klog.Errorf("Error when dumping flows from conntrack: %v", err)
 		return nil, err
 	}
+
+	// Update the antrea_agent_conntrack_max_connection_count prometheus metric
+	maxConns, err := sysctl.GetSysctlNet("nf_conntrack_max")
+	if err != nil {
+		klog.Errorf("Error when getting nf_conntrack_max: %v", err)
+		return nil, err
+	}
+	metrics.MaxConnectionsInConnTrackTable.Set(float64(maxConns))
+
+	metrics.TotalConnectionsInConnTrackTable.Set(float64(len(conns)))
```

Comment:
I was wondering if we could avoid updating metrics in platform-dependent code altogether, by adding a new return value to the `DumpFlows` method in the `ConnTrackDumper` interface:
```go
type conntrackOccupancy struct {
    maxConnections int
    totalConnections int
}

type ConnTrackDumper interface {
    DumpFlows(zoneFilter uint16) ([]*flowexporter.Connection, conntrackOccupancy, error)
}
```

then we can update the metrics in platform-independent code (connections.go)

just an idea, please let me know what you think

Reply:
Thanks, it's a good idea especially considering we will add Windows support for flow exporter in the future. I rewrite the unit tests too after changing the interface in this way.